File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        hashMap = {
            ')': '(',
            ']': '[',
            '}': '{'
        }
        for char in s:
            if char in hashMap:
                if len(stack) and stack[-1]==hashMap[char]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(char)
        if len(stack) == 0:
            return True
        else:
            return False

File: Data Structures & Algorithms/car-fleet/submission-1.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        Deque<Double> stack = new ArrayDeque<>();
        Integer[] order = new Integer[position.length];

        for (int i = 0; i < position.length; i++) {
            order[i] = i;
        }

        Arrays.sort(
            order,
            (a, b) -> Integer.compare(position[b], position[a])
        );

        for (int j = 0; j < order.length; j++) {
            double time = (double) (target - position[order[j]]) / speed[order[j]];

            if (stack.isEmpty() || time > stack.peek()) {
                stack.push(time);
            }
        }
        return stack.size();
    }
}
